hack/generator/pkg/functions/resource_status_setter_function.go

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"fmt"
+
+	"github.com/dave/dst"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astbuilder"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+)
+
+// ResourceStatusSetterFunction is a function that sets the status on a resource
+type ResourceStatusSetterFunction struct {
+	nameOfStatusType string
+	idFactory        astmodel.IdentifierFactory
+}
+
+var _ astmodel.Function = &ResourceStatusSetterFunction{}
```

Comment:
Given the function is non-trivial, I wanted to keep it separate so that I could write a Golden test to verify that it works properly.

I guess I could have put it into this file but reused ObjectFunction from `NewResourceStatusSetterFunction()` though.